Week7/CatApi/app/src/main/java/com/example/catapi/networking/NetworkStatusChecker.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.catapi.networking
+
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+class NetworkStatusChecker(private val connectivityManager: ConnectivityManager?){
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    inline fun performIfConnectedToInternet(action: () -> Unit) : Boolean {
+        return if (hasInternetConnection()) {
+            action()
+            true
+        }else
+            false
+    }
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    fun hasInternetConnection(): Boolean {
```

Comment:
Nice work checking for internet connectivity, I did notice that if there is no connectivity the app hangs with a loading bar. You could cache the breed list in a database table for offline usage although that wouldn't help them with the pictures.